tests/console/yast2_vnc.pm

```diff
@@ -7,88 +7,87 @@
+    # Firewall Settings
+    assert_screen 'yast2_vnc_firewall_port_closed';
+    send_key $firewall_settings{open_port};
+    # Firewall Details
+    assert_screen 'yast2_vnc_firewall_port_open';
+    send_key $firewall_settings{details};
+    assert_screen 'yast2_vnc_firewall_port_details';
+    send_key $firewall_details{network_interfaces};
+    assert_screen 'yast2_vnc_firewall_details_interface_selected';
+    send_key $cmd{ok};
+    assert_screen 'yast2_vnc_firewall_details_selected';
+    # Confirm configuration
+    send_key $confirm;
     assert_screen 'yast2_vnc_warning_text';
-    send_key 'alt-o';
+    send_key $cmd{ok};
     wait_serial('yast2-remote-status-0', 60) || die "'yast2 remote' didn't finish";
-    return if check_var('ARCH', 's390x');    # exit here as port is already open for s390x
+}
 
-    # Check service listening
-    my $cmd_check_port = $older_products ? 'netstat' : 'ss' . ' -tln | grep -E LISTEN.*:5901';
+sub check_service_listening {
+    my $cmd_check_port = $is_newer_product ? 'ss' . ' -tln | grep -E LISTEN.*:5901' : 'netstat';
     if (script_run($cmd_check_port)) {
         record_soft_failure 'boo#1088646 - service vncmanager is not started automatically';
         systemctl('status vncmanager', expect_false => 1);
         systemctl('restart vncmanager');
         systemctl('status vncmanager');
         assert_script_run $cmd_check_port;
     }
-    # Check firewall open for vnc
-    if ($self->firewall eq 'firewalld') {
-        my $cmd_check_firewall = 'firewall-cmd --list-services | grep \'tigervnc tigervnc-https\'';
-        if (script_run($cmd_check_firewall)) {
-            record_soft_failure 'boo#1088647 - firewalld does not create rule for vnc';
-            assert_script_run('firewall-cmd --zone=public --add-service=tigervnc --permanent');
-            assert_script_run('firewall-cmd --zone=public --add-service=tigervnc-https --permanent');
-            assert_script_run('firewall-cmd --reload');
-            assert_script_run($cmd_check_firewall);
-        }
+}
+
+sub check_firewall_open_vnc {
+    my $cmd_check_firewall = 'firewall-cmd --list-services | grep \'tigervnc tigervnc-https\'';
+    if (script_run($cmd_check_firewall)) {
+        record_soft_failure 'boo#1088647 - firewalld does not create rule for vnc';
+        assert_script_run('firewall-cmd --zone=public --add-service=tigervnc --permanent');
+        assert_script_run('firewall-cmd --zone=public --add-service=tigervnc-https --permanent');
+        assert_script_run('firewall-cmd --reload');
+        assert_script_run($cmd_check_firewall);
     }
 }
 
+sub run {
+    my $self = shift;
+
+    select_console 'root-console';
+    assert_script_run("if ! systemctl -q is-active network; then systemctl -q start network; fi");
+    zypper_call 'in vncmanager xorg-x11' . (!$is_newer_product ? ' net-tools' : '');
+    script_run("yast2 remote; echo yast2-remote-status-\$? > /dev/$serialdev", 0);
```

Comment:
Just for consistency, I like this approach to keep run method clean and clear, so I would move 4 lines above to `test_setup` subroutine, for example.